app/Http/Controllers/PlaylistController.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Playlist;
+use App\Song;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class PlaylistController extends Controller
+{
+    /**
+     * Create a new playlist.
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     * @throws \Illuminate\Validation\ValidationException
+     */
+    public function create(Request $request)
+    {
+        $this->validate($request, [
+            'name' => 'required',
+        ]);
+
+        $playlist = new Playlist();
+        $playlist->name = $request->name;
+        $playlist->is_private = 0;
+        $playlist->user_id = Auth::user()->id;
+
+        $playlist->save();
+
+        return response()->json([
+            'success' => true,
+            'playlist' => $playlist
+        ]);
+    }
+
+    /**
+     * Add song to a playlist.
+     *
+     * @param Playlist $playlist_id
+     * @param Song $id
+     * @return \Illuminate\Http\JsonResponse
+     * @throws \Illuminate\Validation\ValidationException
+     */
+    public function add($playlist_id, $id)
+    {
+        $playlist = Playlist::find($playlist_id);
+
+        if ((string)Auth::user()->id !== $playlist->user_id)
```

Comment:
We should probably move this kind of checks into gates/policies.

See: https://laravel.com/docs/5.8/authorization